cpp/oneapi/dal/algo/knn/backend/gpu/infer_kernel_impl_dpc_distr.hpp

```diff
@@ -558,6 +495,13 @@ sycl::event bf_kernel_distr(sycl::queue& queue,
 
     comm.allgather(tcount, node_sample_counts.flatten()).wait();
 
+    // auto [max_tcount, _] = pr::argmax(queue, node_sample_counts);
+    std::int64_t max_tcount = 0;
+    for (std::int32_t index = 0; index < node_sample_counts.get_count(); ++index) {
```

Comment:
```suggestion
    for (std::int64_t index = 0; index < node_sample_counts.get_count(); ++index) {
```

Reply:
This loop iterates through each device to find highest number of train data on any GPU. I chose int32 since I don't think number of devices would ever exceed ~thousands, let alone int32 max. But if suggestion is for uniformity then I can change it.